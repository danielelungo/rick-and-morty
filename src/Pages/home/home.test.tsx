/* eslint-disable react/react-in-jsx-scope */
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Home from "./home";

const MockHome = () => {
  return (
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  );
};

describe("Home component", () => {
  test("renders Home Page", () => {
    const tree = renderer.create(<MockHome />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("render home text", () => {
    render(<MockHome />);
    const headingElement = screen.getByText(/home/i);
    expect(headingElement).toBeInTheDocument();
  });
});
