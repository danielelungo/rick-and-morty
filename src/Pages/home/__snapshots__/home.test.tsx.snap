// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders Home Page 1`] = `
Array [
  <h1>
    Home
  </h1>,
  <div
    className="loader"
    data-testid="oval-loading"
    style={
      Object {
        "display": "flex",
        "padding": 3,
      }
    }
  >
    <svg
      aria-label="oval-loading"
      data-testid="oval-svg"
      height={150}
      stroke="#4fa94d"
      viewBox="-20 -20 42 42"
      width={150}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          strokeWidth={2}
          transform="translate(1 1)"
        >
          <circle
            cx="0"
            cy="0"
            r={20}
            stroke="#4fa94d"
            strokeOpacity=".5"
            strokeWidth={2}
          />
          <path
            d="M20 0c0-9.94-8.06-20-20-20"
          >
            <animateTransform
              attributeName="transform"
              dur="1s"
              from="0 0 0"
              repeatCount="indefinite"
              to="360 0 0"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>,
]
`;
